Frame.cc

```diff
@@ -961,16 +961,20 @@ bool Frame::FillSpectralProfileData(
                         profile_ok = true;
                         return profile_ok;
                     }
+                    bool region_mask_ok(false);
                     bool use_swizzled_data(false);
+                    std::unique_lock<std::mutex> guard(_image_mutex);
                     try {
                         // check is the region mask valid (outside the lattice or not)
                         region->XyMask();
+                        region_mask_ok = true;
                         // if region mask is valid, then check is swizzled data available
-                        std::unique_lock<std::mutex> guard(_image_mutex);
                         use_swizzled_data = _loader->UseRegionSpectralData(region->XyMask());
-                        guard.unlock();
```

Comment:
ideally, we should only lock while getting the Xy mask, and then unlock, rather than locking while doing the entire maks check

Reply:
Called again instead of reusing mask:
`_loader->GetRegionSpectralData(profile_stokes, region_id, region->XyMask(), region->XyOrigin(),`